runtime/compiler/trj9/runtime/J9Profiler.cpp

```diff
@@ -1932,6 +1932,25 @@ TR_BlockFrequencyInfo::getRawCount(TR::ResolvedMethodSymbol *resolvedMethod, TR_
    }
 
 int32_t
+TR_BlockFrequencyInfo::getOriginalBlockNumberToGetLiveFrequency(TR_ByteCodeInfo &bci, TR::Compilation *comp)
+   {
+   int32_t callerIndex = bci.getCallerIndex();
+   TR::ResolvedMethodSymbol *resolvedMethod = callerIndex < 0 ? comp->getMethodSymbol() : comp->getInlinedResolvedMethodSymbol(callerIndex);
+   int32_t byteCodeToSearch = resolvedMethod->getProfilingByteCodeIndex(bci.getByteCodeIndex());
+   TR_ByteCodeInfo searchBCI = bci;
+   searchBCI.setByteCodeIndex(byteCodeToSearch); 
+   for (auto i=0; i < _numBlocks; ++i)
+      {
+      if (_callSiteInfo->hasSameBytecodeInfo(_blocks[i], searchBCI, comp)
+         || (_blocks[i].getCallerIndex() == searchBCI.getCallerIndex() && _blocks[i].getByteCodeIndex() == searchBCI.getByteCodeIndex()))
```

Comment:
so this is a general API so the bci may not match a block start so this needs consider the range of offsets a given block covers. I am still checking to make sure the callerindex is safe across recompilation uses. The name should not include the word frequency since profiling related code frequency refers to the normalized frequency between 6 and 10000 - this should use the term raw count or similar to avoid confusion.

This needs some Javadoc of some kind as well.